src/analysis/retail/shaman/enhancement/modules/talents/ElementalBlastGuide.tsx

```diff
@@ -0,0 +1,323 @@
+      this.onUpdateSpellUsable,
+    );
+  }
+
+  onDamage(event: DamageEvent) {
+    this.cast!.bonusDamage = calculateEffectiveDamage(
+      event,
+      Math.pow(1.2, this.cast?.elementalSpiritsActive ?? 0),
+    );
+    this.recordCooldown(this.cast!);
+    this.cast = null;
+  }
+
+  onCast(event: CastEvent) {
+    const cr = getResource(event.classResources, RESOURCE_TYPES.MAELSTROM_WEAPON.id);
+    this.cast = {
+      chargesBeforeCast: this.currentCharges,
+      elementalSpiritsActive: this.activeElementalSpirits,
+      maelstromUsed: cr?.cost ?? 0,
+      event: event,
+      bonusDamage: 0,
+    };
+    this.currentCharges -= 1;
+  }
+
+  onUpdateSpellUsable(event: UpdateSpellUsableEvent) {
+    if (
+      event.updateType === UpdateSpellUsableType.RestoreCharge ||
+      event.updateType === UpdateSpellUsableType.EndCooldown
+    ) {
+      this.currentCharges += 1;
+    }
+  }
+
+  onApplyElementalSpiritBuff(event: ApplyBuffEvent) {
+    this.elementalSpritsActive[event.ability.guid] += 1;
+  }
+
+  onRemoveElementalSpiritBuff(event: RemoveBuffEvent) {
+    this.elementalSpritsActive[event.ability.guid] -= 1;
+    if (this.elementalSpritsActive[event.ability.guid] < 0) {
+      this.elementalSpritsActive[event.ability.guid] = 0;
+    }
+  }
+
+  get activeElementalSpirits(): number {
+    return Object.keys(this.elementalSpritsActive).reduce(
+      (total: number, buffId: string) => (total += this.elementalSpritsActive[Number(buffId)]),
+      0,
+    );
+  }
+
+  getOverallCastPerformance(cast: ElementalBlastCastDetails) {
+    if (cast.chargesBeforeCast === 2 && cast.maelstromUsed >= 5) {
+      return QualitativePerformance.Perfect;
+    } else {
+      switch (cast.elementalSpiritsActive) {
+        case 0:
+          /** elemental blast should not be cast without elemental spirits if uncapped on charges */
+          return QualitativePerformance.Fail;
+        case 1:
+          return cast.maelstromUsed === 10
+            ? QualitativePerformance.Good
+            : QualitativePerformance.Ok;
+        default:
+          return cast.maelstromUsed >= 8
+            ? QualitativePerformance.Perfect
+            : QualitativePerformance.Good;
+      }
+    }
+  }
+
+  description(): JSX.Element {
+    return (
+      <>
+        <SpellLink spell={TALENTS.ELEMENTAL_BLAST_ELEMENTAL_TALENT} /> damage is multiplicatively
+        increased for each <SpellLink spell={TALENTS.ELEMENTAL_SPIRITS_TALENT} /> that is currently
+        active.
+      </>
+    );
+  }
+
+  maelstromPerformance(cast: ElementalBlastCastDetails): UsageInfo {
+    return {
+      performance:
+        cast.maelstromUsed === 10
+          ? QualitativePerformance.Perfect
+          : cast.maelstromUsed >= 8
+            ? QualitativePerformance.Good
+            : cast.maelstromUsed >= 5
+              ? QualitativePerformance.Ok
+              : QualitativePerformance.Fail,
```

Comment:
my personal pref is to do if/else for this. not a huge fan of (deeply) nested ternaries. but up to you

Reply:
I'm not either, I'll fix this